pkg/storage/snapshot/restore.go

```diff
@@ -995,7 +995,12 @@ func CreateRestorePVCDef(restorePVCName string, volumeSnapshot *vsv1.VolumeSnaps
 			Labels:      sourcePVC.Labels,
 			Annotations: sourcePVC.Annotations,
 		},
-		Spec: sourcePVC.Spec,
+		Spec: corev1.PersistentVolumeClaimSpec{
+			AccessModes:      sourcePVC.Spec.AccessModes,
+			Resources:        sourcePVC.Spec.Resources,
+			StorageClassName: sourcePVC.Spec.StorageClassName,
+			VolumeMode:       sourcePVC.Spec.VolumeMode,
+		},
```

Comment:
Given this new pattern of explicitly setting the fields we care about I think it makes sense to set dataSource and dataSourceRef in in this block as well

Reply:
@muxuelanKK you planning to come back to this?